File: ios/Sources/ConfigReaderPlugin/ConfigReaderPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(ConfigReaderPlugin)
public class ConfigReaderPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "ConfigReaderPlugin"
    public let jsName = "ConfigReader"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "readInfoPlistValue", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = ConfigReader()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
    
    @objc func readInfoPlistValue(_ call: CAPPluginCall) {
        guard let key = call.getString("key") else {
            call.reject("Key parameter is required")
            return
        }
        
        let value = implementation.readInfoPlistValue(key)
        call.resolve([
            "value": value
        ])
    }
}

File: android/src/main/java/ooo/latch/app/ConfigReaderPlugin.java
package technology.mumford.capacitor-config-reader;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "ConfigReader")
public class ConfigReaderPlugin extends Plugin {

    private ConfigReader implementation = new ConfigReader();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }

    @PluginMethod
    public void readStringResource(PluginCall call) {
        String key = call.getString("key");
        
        if (key == null || key.isEmpty()) {
            call.reject("Key parameter is required");
            return;
        }

        String value = implementation.readStringResource(getContext(), key);
        
        JSObject ret = new JSObject();
        ret.put("value", value);
        call.resolve(ret);
    }
}
